paramiko/proxy.py

```diff
@@ -119,7 +120,22 @@ def recv(self, size):
             raise ProxyCommandFailure(" ".join(self.cmd), e.strerror)
 
     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        # Before killing a process, it is necessary to check if it exists
+        # Otherwise, an error will be reported during asynchronous calls
+        if self.process.poll() is None:
+            try:
+                # close resources first
+                self.process.stdin.close()
+                self.process.stdout.close()
+                self.process.stderr.close()
```

Comment:
`close()`  is redundant, I have deleted it.
But it is necessary to check whether a process is still present before killing it.

Because when I use distributed task queues [Celery](https://docs.celeryq.dev/en/stable/index.html), this process is usually killed by the system.
It is possible to report an error `ProcessLookupError`.

In short, it is necessary to check whether a process is still alive before killing it.

Reply:
I'm sorry, I think I expressed myself poorly -- I wasn't requesting that you change your implementation, only that you add some comments &c. to document it more thoroughly.

Although, if that `if self.closed:` check will do just as well as the full set of stream `.close()` calls and the `try`/`except`, that's certainly much cleaner.

Reply:
I added some comments before the code, do you think it's feasible?

Reply:
There is no need to close IO as it will kill directly

Reply:
Before killing this subprocess, you need to determine whether it exists